### PR TITLE
fixes #36454, added properties to stripe API

### LINKED ISSUE #36454: [@types/stripe] 6.30.3 requested_capabilities missing in create account
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/string` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @wjohnsto @CodeAnimal @sampsonjoliver @LinusU @brannon @kkamperschroer @starhoshi @bruun @galtalmor @htunnicliff @squirly @tzarger @ifiokjr @yultyyev @cpsoinos @adamduren @saranshkataria @0xJoKe @delianides @gokulchandra @viralpickaxe @chrisleck @spacetag 

If you do not mention the authors the issue will be ignored.

When I try to create a Connect Account it does not recognize the requested_capabilities option (I get an error, see below) however this is a required option (https://stripe.com/docs/upgrades#2019-02-19).

ERROR: `Object literal may only specify known properties, and 'requested_capabilites' does not exist in type 'IAccountCreationOptions'`.

To reproduce:

```typescript
/**
 * "@types/stripe": "^6.30.3"
 * "stripe": "^7.3.0"
 */


import * as stripe from 'stripe';

const client = new stripe('mykey');
client.setApiVersion('2019-05-16');

const dob = new Date(1980, 1, 1);
(async () => {
  const act = await client.accounts.create({
    email: 'jd@email.com',
    country: 'US',
    type: 'custom',
    business_profile: {
      name: 'John Doe',
      support_email: 'jd@gmail.com',
      support_phone: '1231231234',
      support_url: 'https://mtwebsite.com',
      product_description: 'My Description'
    },
    metadata: {
      id: '123abc'
    },
    business_type: 'individual',
    individual: {
      address: {
        city: 'Seattle',
        country: 'US',
        line1: '123 Main St',
        line2: 'UNIT 1',
        postal_code: '123456',
        state: 'WA'
      },
      dob: {
        day: dob.getDate(),
        month: dob.getMonth(),
        year: dob.getFullYear()
      },
      first_name: 'John',
      last_name: 'Doe',
      phone: '1231231234'
    },
    requested_capabilities: ['platform_payments'], // <= ERROR
    tos_acceptance: {
      date: Math.floor(Date.now() / 1000),
      ip: '8.8.8.8'
    }
  });
})();
```

For now, I have been able to use this work around:

```
  const opts: stripe.account.IAccountCreationOptions = {
    email: 'jd@email.com',
    country: 'US',
    type: 'custom',
    business_profile: {
      name: 'John Doe',
      support_email: 'jd@gmail.com',
      support_phone: '1231231234',
      support_url: 'https://mtwebsite.com',
      product_description: 'My Description'
    },
    metadata: {
      id: '123abc'
    },
    business_type: 'individual',
    individual: {
      address: {
        city: 'Seattle',
        country: 'US',
        line1: '123 Main St',
        line2: 'UNIT 1',
        postal_code: '123456',
        state: 'WA'
      },
      dob: {
        day: dob.getDate(),
        month: dob.getMonth(),
        year: dob.getFullYear()
      },
      first_name: 'John',
      last_name: 'Doe',
      phone: '1231231234'
    },
    tos_acceptance: {
      date: Math.floor(Date.now() / 1000),
      ip: '8.8.8.8'
    }
  };

  opts['requested_capabilities'] = ['platform_payments']

  const act = await client.accounts.create(opts);
```

However, I think this is something that really needs to be resolved quickly, as this is a required option now.

### PR DESCRIPTION
**Pull request by saranshkataria**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by saranshkataria** [comment]
This fixes #36454

**Comment by esetnik** [comment]
@saranshkataria are you actively working on this fix still?

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -1039,6 +1046,31 @@ declare namespace Stripe {
...
+                details?: string;
+
+                /**
+                 * One of document_address_mismatch, document_dob_mismatch, document_duplicate_type, document_id_number_mismatch,
+                 * document_name_mismatch, failed_keyed_identity, or failed_other.
+                 * A machine-readable code specifying the verification state for the person.
+                 */
+                details_code?:
+                    | 'document_address_mismatch'
+                    | ' document_dob_mismatch'
```
Comment (RyanCavanaugh) [inline types/stripe/index.d.ts]: You have leading spaces *inside* the string literal for several of these.

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/stripe/index.d.ts (hunk 1 above)
Comment (saranshkataria) [reply types/stripe/index.d.ts]: good catch, resolving conflicts caused this I suppose, pushing a fix

**Pull request merged**